packages/eslint-plugin/tests/rules/no-misused-promises.test.ts

```diff
@@ -500,780 +500,1768 @@ foo(bar);
       },
       options: [{ checksVoidReturn: { attributes: true } }],
     },
-  ],
+    {
+      // Valid void-returning class
```

Comment:
Nit: rule tests have a bad habit of drifting away from their commented intent over time. The tests you're adding are pretty clear & apparent. I'd suggest removing these comments.

```suggestion
```

Reply:
Ah ok, the massive list of test cases without sections/etc still scares me but I trust you 😁  removed!